raiden/ui/startup.py

```diff
@@ -0,0 +1,361 @@
+
+import click
+from eth_utils import to_canonical_address, to_checksum_address
+
+from raiden.constants import Environment, RoutingMode
+from raiden.exceptions import AddressWithoutCode, AddressWrongContract, ContractVersionMismatch
+from raiden.network.blockchain_service import BlockChainService
+from raiden.network.discovery import ContractDiscovery
+from raiden.network.pathfinding import configure_pfs
+from raiden.network.proxies import (
+    SecretRegistry,
+    ServiceRegistry,
+    TokenNetworkRegistry,
+    UserDeposit,
+)
+from raiden.network.throttle import TokenBucket
+from raiden.network.transport import UDPTransport
+from raiden.settings import DEVELOPMENT_CONTRACT_VERSION, RED_EYES_CONTRACT_VERSION
+from raiden.utils.typing import Address
+from raiden_contracts.constants import (
+    CONTRACT_ENDPOINT_REGISTRY,
+    CONTRACT_SECRET_REGISTRY,
+    CONTRACT_SERVICE_REGISTRY,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    CONTRACT_USER_DEPOSIT,
+    ID_TO_NETWORKNAME,
+)
+from raiden_contracts.contract_manager import (
+    contracts_precompiled_path,
+    get_contracts_deployment_info,
+)
+
+from .sync import check_discovery_registration_gas
+
+
+def environment_type_to_contracts_version(environment_type: Environment) -> str:
+    if environment_type == Environment.DEVELOPMENT:
+        contracts_version = DEVELOPMENT_CONTRACT_VERSION
+    else:
+        contracts_version = RED_EYES_CONTRACT_VERSION
+
+    return contracts_version
+
+
+def setup_network_id_or_exit(
+        config: Dict[str, Any],
+        given_network_id: int,
+        web3,
+) -> Tuple[int, bool]:
+    """
+    Takes the given network id and checks it against the connected network
+
+    If they don't match, exits the program with an error. If they do adds it
+    to the configuration and then returns it and whether it is a known network
+    """
+    node_network_id = int(web3.version.network)  # pylint: disable=no-member
+    known_given_network_id = given_network_id in ID_TO_NETWORKNAME
+    known_node_network_id = node_network_id in ID_TO_NETWORKNAME
+
+    if node_network_id != given_network_id:
+        if known_given_network_id and known_node_network_id:
+            click.secho(
+                f"The chosen ethereum network '{ID_TO_NETWORKNAME[given_network_id]}' "
+                f"differs from the ethereum client '{ID_TO_NETWORKNAME[node_network_id]}'. "
+                "Please update your settings.",
+                fg='red',
+            )
+        else:
+            click.secho(
+                f"The chosen ethereum network id '{given_network_id}' differs "
+                f"from the ethereum client '{node_network_id}'. "
+                "Please update your settings.",
+                fg='red',
+            )
+        sys.exit(1)
+
+    config['chain_id'] = given_network_id
+    return given_network_id, known_node_network_id
+
+
+def setup_environment(config: Dict[str, Any], environment_type: Environment) -> Environment:
```

Comment:
Why not set the contracts version here as well? This seems quite similar to `environment_type_to_contracts_version`

Reply:
They are setting different things up. The `setup_contracts_or_exit()` function sets the contracts version, and defines their deployment data depending on network id and environment type.

The `setup_environment` version simply defines what kind of environment is selected by the user and sets up any environment specific options such as the private matrix rooms.

They could be combined as functions but the whole point of this imo is to split this into smaller logical units.